tests/test_compiler.py

```diff
@@ -206,3 +206,80 @@ def circuit(x: float):
         mlir_str = str(circuit.mlir)
         result_header = "func.func private @circuit(%arg0: tensor<f64>) -> tensor<f64>"
         assert result_header in mlir_str
+
+
+class TestCatalystMeasure:
+    """Test ``qml.measure`` in JIT mode with Catalyst"""
+
+    def test_measure_basic(self):
+        """Test measure (basic)."""
+        dev = qml.device("lightning.qubit", wires=1)
+
+        @qml.qjit
+        @qml.qnode(dev)
+        def circuit(x: float):
+            qml.RX(x, wires=0)
+            return qml.measure(wires=0)
```

Comment:
this is quite notable imo. this circuit would fail without the qjit decorator because it returns a `MeasurementValue`, which is not a valid `MeasurementProcess`. I'm not sure how we feel about catalyst enabling things that are otherwise not allowed in PennyLane - do we want to augment PL, block this even though catalyst allows it, or leave it as-is?

Reply:
@timmysilv I'm okay with this divergence for now, but we aim to bring PL and Catalyst more inline in 2024.

So for now there will be some things that work with `@qjit` and not without (postprocessing arbitrary measurements, elifs, etc.), some things that work _without_ qjit but not with (pulse programming, `qml.exp`, certain transforms, etc.), and also cases where the same code _works in both cases but behaves differently_.

My philosophy is to allow this for now, but we want to address it over time.

Reply:
Oh wait, I didn't realize what PR I was commenting in 😆 

This PR is a bit different, since the same code _works in both cases but behaves differently_ (that is, mid-circuit measurements with statistics behaves differently in PL and catalyst currently).

So @trbromley and I have elected not to merge this one in for now until we come up with a better solution.